core/src/serve_repair.rs

```diff
@@ -258,6 +258,62 @@ pub enum RepairProtocol {
     },
 }
 
+const REPAIR_REQUEST_PONG_SERIALIZED_BYTES: usize = PUBKEY_BYTES + HASH_BYTES + SIGNATURE_BYTES;
+const REPAIR_REQUEST_HEADER_SERIALIZED_BYTES: usize =
+    SIGNATURE_BYTES + PUBKEY_BYTES + PUBKEY_BYTES + 8 + 4;
+
+fn check_well_formed_repair_request(packet: &Packet, stats: &mut ServeRepairStats) -> Result<()> {
+    if let Some(discriminator) = packet
+        .data(..4)
+        .and_then(|data| <[u8; 4]>::try_from(data).ok())
+        .map(u32::from_le_bytes)
+    {
+        let expected_size = match discriminator {
+            0..=6 => {
+                // deprecated requests
+                stats.err_unsigned += 1;
+                return Err(Error::from(RepairVerifyError::Unsigned));
```

Comment:
If we hit this case on MNB or Devnet, we will count it as well formed independent of what the size is. Is that the intention?

Reply:
discussed offline